crates/uv/tests/pip_install.rs

```diff
@@ -3008,3 +3008,421 @@ fn deptry_gitignore() -> Result<()> {
 
     Ok(())
 }
+
+/// Install an editable package that depends on a previously installed editable package.
+#[test]
+fn dependent_editable_previously_installed() -> Result<()> {
+    let context = TestContext::new("3.12");
+    let root_path = current_dir()?.join("../../scripts/packages/dependent_editables");
+
+    // In addition to the standard filters, remove the temporary directory from the snapshot.
+    let filters: Vec<_> = [(r"file://.*/", "file://[TEMP_DIR]/")]
+        .into_iter()
+        .chain(INSTA_FILTERS.to_vec())
+        .collect();
+
+    // Install the first editable
+    uv_snapshot!(filters, command(&context)
+        .arg("-e")
+        .arg(root_path.join("first_editable")), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Built 1 editable in [TIME]
+    Resolved 1 package in [TIME]
+    Installed 1 package in [TIME]
+     + first-editable==0.0.1 (from file://[TEMP_DIR]/first_editable)
+    "###
+    );
+
+    // The second editable depends on the first editable, which is only available
+    // locally and should not be fetched from an index
+    uv_snapshot!(filters, command(&context)
+        .arg("-e")
+        .arg(root_path.join("second_editable")), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Built 1 editable in [TIME]
+    Resolved 2 packages in [TIME]
+    Installed 1 package in [TIME]
+     + second-editable==0.0.1 (from file://[TEMP_DIR]/second_editable)
+    "###
+    );
+
+    Ok(())
+}
+
+/// Install an editable package that depends on a previously installed editable package.
+/// Request reinstall of the first package.
+#[test]
+fn dependent_editable_previously_installed_reinstall() -> Result<()> {
+    let context = TestContext::new("3.12");
+    let root_path = current_dir()?.join("../../scripts/packages/dependent_editables");
+
+    // In addition to the standard filters, remove the temporary directory from the snapshot.
+    let filters: Vec<_> = [(r"file://.*/", "file://[TEMP_DIR]/")]
+        .into_iter()
+        .chain(INSTA_FILTERS.to_vec())
+        .collect();
```

Comment:
These filters need clean-up, copied from Charlie's test case but it's not a temporary directory it's the workspace